astropy/io/registry/registry.py

```diff
@@ -0,0 +1,589 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import os
+import re
+import sys
+import json
+
+import numpy as np
+
+from ...utils import OrderedDict
+from ...extern import six
+from ...extern.six.moves import zip
+
+__all__ = ['identify_format', 'get_reader', 'get_writer', 'read', 'write',
+           'get_formats', 'BaseIO']
+
+__doctest_skip__ = ['register_identifier']
+
+
+_registry = OrderedDict()
+
+def _load_builtins():
+    global _registry
+    with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), "registry.json")) as f_reg:
+        _registry.update(json.load(f_reg))
+
+# Read in registry of built-in formats
+_load_builtins()
+
+
+
+def _get_object_by_string(path_to_function):
+    """
+    Given a path, e.g. 'astropy.table.table.Table', return the actual object at
+    that path.
+    """
+    module, function = path_to_function.rsplit('.', 1)
+    if sys.version_info[:2] < (2, 7):
+        __import__(module)
+        return getattr(sys.modules[module], function)
+    else:
+        from importlib import import_module
+        return getattr(import_module(module), function)
+
+
+def _get_object_name(object_class):
+    """
+    Given either a class or a string giving the path to a class, return the
+    name of the class.
+    """
+    if isinstance(object_class, six.string_types):
+        return object_class.split('.')[-1]
+    else:
+        return object_class.__name__
+
+
+def get_registry_subset(data_class, identify=None, read=None, write=None,
+                        data_format=None, resolve=False):
+    """
+    Return a subset of the registry.
+
+    Parameters
+    ----------
+    data_class : class
+        The class to return a subset of the registry for. If this class matches
+        both a class in the registry as well as another class that it is a
+        subclass of, the formats that match the class directly take precedence
+        over the parent classes.
+    identify : bool
+        If set, only include the subset of the registry that have the identify
+        flag set to this value.
+    read : bool
+        If set, only include the subset of the registry that have the read
+        flag set to this value.
+    write : bool
+        If set, only include the subset of the registry that have the write
+        flag set to this value.
+    data_format : str
+        If set, only include the subset of the registry for this format.
+    resolve : bool
```

Comment:
It looks like this keyword is being ignored.  I would have expected this at least to be used in front of line 132 at the end to decide whether to resolve IO classes.